### PR TITLE
Replacing Roboto with Inter

### PR DESCRIPTION
**Pull request by dialpad-drew**
## Description
This PR removes Roboto from Dialtone and replaces it with Inter. It also removes the thin, light, medium, and black font-weight files.

Also updates font-weight utility class documentation.

**Note:**
* I looked into using the variable font file, but was going to have to touch the mixin and wasn't confident in that approach. Simply using the individual weight files seemed like the quickest win.
* I kept the removed font-weight variables and utility classes as shims, so they'll still worked for the time-being. Removing them altogether would have removed their ability to gracefully degrade.
* Any font-weight style (derived either from a variable or utility class) 500 or below will be rendered as 400
* Any font-weight style (derived either from a variable or utility class) 600 or above will be rendered as 700

### CODE REVIEW

File: lib/dist/css/dialtone.css [hunk 1]
```diff
@@ -36041,7 +36029,7 @@ body {
   --grid-columns: 12;
   --grid-column-width: minmax(min-content, 24rem);
   --grid-column-auto: 1fr;
-  --ff-custom: "Roboto", -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
+  --ff-custom: "Inter", -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
   --ff-sans: -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
   --ff-mono: "RobotoMono", "SFMono-Regular", Consolas, "Liberation Mono", Menlo, Courier, monospace;
```
Comment (ryandialpad) [inline lib/dist/css/dialtone.css:36034]: @dialpad-drew Does this line need to be updated as well to remove roboto?

File: lib/dist/css/dialtone.css [hunk 2]
```diff
@@ -35260,28 +35260,16 @@ body.theme-dark .d\:d-bgg-to-0 {
...
 @font-face {
-  font-family: "Roboto";
+  font-family: "Inter";
   font-style: "normal";
   font-weight: 700;
-  src: url("../fonts/Roboto-Bold.woff2") format("woff2");
+  src: url("../fonts/Inter-Bold.woff2") format("woff2");
 }
 @font-face {
   font-family: "RobotoMono";
```
Comment (ryandialpad) [inline lib/dist/css/dialtone.css:35275]: @dialpad-drew Does this line need to be updated as well to remove roboto?

File: lib/build/less/variables/typography.less [hunk 3]
```diff
@@ -17,17 +17,15 @@
 //  ============================================================================
 //  $   FONT FAMILY
 //  ----------------------------------------------------------------------------
-@define-ff-custom:               @ff-roboto;
-@define-ff-custom-name:          "Roboto";
+@define-ff-custom:               @ff-inter;
+@define-ff-custom-name:          "Inter";
 @define-ff-mono:                 @ff-roboto-mono;
 @define-ff-mono-name:            "RobotoMono";
```
Comment (ryandialpad) [inline lib/build/less/variables/typography.less:23]: This probably answers my previous question, do we still need this?

cc @dialpad-drew

**Comment by ryandialpad** [comment]
Screen Shot 2021-06-23 at 2 39 04 PM

@dialpad-drew Did we also need to update the docs in this pr?

Screenshot taken from local build of the branch

File: lib/build/less/variables/typography.less (hunk 3 above)
Comment (dialpad-drew) [reply lib/build/less/variables/typography.less:23]: Inter doesn't have a mono font-family and we need something for showing `our various code snippets`, so figured I'd keep it.

Comment (ryandialpad) [reply lib/build/less/variables/typography.less:23]: Sound good, thanks!

**Comment by dialpad-drew** [comment]
> Screen Shot 2021-06-23 at 2 39 04 PM
> 
> @dialpad-drew Did we also need to update the docs in this pr?
> 
> Screenshot taken from local build of the branch

Good catch. Pushed update.

**Review by ryandialpad — APPROVED** [review]
🚀

**Comment by dbecherdialpad** [comment]
Can you create tickets for the cleanup and improvements you mentioned? (If you didn't already)

**Comment by dialpad-drew** [comment]
> Can you create tickets for the cleanup and improvements you mentioned? (If you didn't already)

https://switchcomm.atlassian.net/browse/DP-38255

**Comment by dbecherdialpad** [comment]
@dialpad-drew ty. Did we want to investigate using the variable font file in the future? That seems like it would be a nice performance win.

**Comment by dialpad-drew** [comment]
> @dialpad-drew ty. Did we want to investigate using the variable font file in the future? That seems like it would be a nice performance win.

Sure did, fam. It's in the backlog. 🙌

**Pull request merged**